src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -426,7 +427,7 @@ void ReadFromSystemZooKeeper::applyFilters()
 
 void ReadFromSystemZooKeeper::fillData(MutableColumns & res_columns)
 {
-    zkutil::ZooKeeperPtr zookeeper = context->getZooKeeper();
+    QueryStatusPtr query_status = context->getProcessListElement();
```

Comment:
Let's use the whole pack:
* Use `insert_keeper_max_retries` and similar for settings. I hope we will unify it with `backup_restore_keeper_max_retries` and have a single common setting for retries at some point, so we shouldn't add a new one.
* Use `ZooKeeperWithFaultInjection` and refresh it in the loop if necessary.